### ISSUE TITLE
crash when clicking 'Enter the Dungeon' (Windows)

### ISSUE THREAD

**Issue by bughunter2** [00-Evan/shattered-pixel-dungeon#481]
When this crash happens, there's no way to start a new game (at least, not without using workarounds).

This problem can occur in two ways:

Method A: Start the game for the first time (so there's no game AppData yet). Then, skip the intro / class selection (via the button in the top right corner). Now click 'Enter the Dungeon'. The game crashes.

Method B: Play a game, but make sure you have only 1 saved game (so it's your last one). Die. Go to the main menu (or quit the game). The next time you click 'Enter the Dungeon', the game crashes.

The crash seems to happen when you click the button while there aren't any existing saved games. I.e., when there doesn't already exist any 'game%d' folder in "%AppData%\.shatteredpixel\Shattered Pixel Dungeon".

A simple workaround for users:
Quit the game. Edit 'settings.xml' and set 'intro' to 'true'. Start the game and pick a class for your new game.

**Comment by 00-Evan** [comment]
I'm not able to reproduce this. The game correctly goes to the 'choose your hero' screen when either method is used. The game should give a popup or some console log data about what prompted the crash, can you share that?

**Comment by bughunter2** [comment]
Yes, I found a log file which I have attached. I'm using Java/JRE version 8 update 261 (build 1.8.0_261-b12)

The `Internal exceptions (10 events)` part seems interesting to me.

[hs_err_pid1920.log](https://github.com/00-Evan/shattered-pixel-dungeon/files/5133259/hs_err_pid1920.log)

**Comment by 00-Evan** [comment]
Hmm, the fact that it's a native crash makes me think it's caused by your OS or JVM. Or is at least being exposed by it.

The only difference between pressing the 'enter the dungeon' button when a game is in progress or not is that if no games are in progress it jumps straight to hero select. There isn't any special logic that happens otherwise, and certainly nothing that should prompt a native crash.

**Comment by bughunter2** [comment]
Seems the interesting line was this:
`# C  [atio6axx.dll+0xbc6dfd]`
Which hints at the AMD driver being the culprit.

I'm getting different results with various driver versions:

I (believe I) was using this driver from 2017 at the time of the crash:
`Non-WHQL-Win7-64Bit-Radeon-Software-Crimson-ReLive-17.11.4-Nov27.exe`

However, the atio6axx.dll file seems to be a leftover from this beta/non-WHQL 2019 driver which I uninstalled about a week ago:
`Win7-Radeon-Software-Adrenalin-2019-Edition-19.12.1-Dec2.exe`

Today, to debug this issue, I have done the following:

I have uninstalled the drivers (and used the `AMD Cleanup Utility` afterwards).
Then, I installed this old driver from 2013:
`13-12_win7_win8_64_dd_ccc_whql.exe`
Result: No crash. Game works.

Afterwards, I uninstalled the old driver (did cleanup as well), and installed the current 2020 driver:
`win7-radeon-software-adrenalin-2020-edition-20.4.2-may25.exe`
Result: No crash. Game works.

Each time I made sure the ".shatteredpixel" directory was the same (no saved games).

In conclusion, it was a driver issue.

Just posting this here in case anybody else runs into AMD driver issues while using Java applications.
Maybe it helps somebody.

**Comment by 00-Evan** [comment]
Thanks for looking into this on your end, glad you were able to resolve it!
